### PR TITLE
[IMP] Management of due cost for Ri.Ba.

### PR DESCRIPTION
**Pull request by andrea4ever**
On invoice validate a new line with due cost was add on invoice

[IMP] Test for new function

### CODE REVIEW

File: l10n_it_ricevute_bancarie/__openerp__.py [hunk 1]
```diff
@@ -84,6 +84,7 @@
         "views/account_view.xml",
         "views/wizard_riba_issue.xml",
         "views/wizard_riba_file_export.xml",
+        "views/account_config_view.xml",
         "riba_workflow.xml",
         # "security/ir.model.access.csv",
```
Comment (OpenCode) [inline l10n_it_ricevute_bancarie/__openerp__.py]: @andrea4ever remove commented line if is useless

File: l10n_it_ricevute_bancarie/models/account/account.py [hunk 2]
```diff
@@ -137,4 +142,99 @@ def _get_invoice_by_move_line(self, cr, uid, ids, context=None):
...
         }
+
+    def _get_first_date_due(self):
+        pterm = self.env['account.payment.term'].browse(self.payment_term.id)
+        pterm_list = pterm.compute(value=1, date_ref=self.date_invoice)[0]
+        return min(line[0] for line in pterm_list)
+
+    def month_check(self, invoice_date_due, all_date_due):
+        """
+
```
Comment (OpenCode) [inline l10n_it_ricevute_bancarie/models/account/account.py]: No empty line in docstring

File: l10n_it_ricevute_bancarie/tests/test_invoice_due_cost.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            'payment_cost': 5.00,
+            'line_ids': [
+                (0, 0,
+                 {'value': 'balance', 'days': 30,
+                  'days2': -1})
+            ]
+        })
+
+    def _account_expense(self):
+        return self.env['account.account'].create({
```
Comment (OpenCode) [inline l10n_it_ricevute_bancarie/tests/test_invoice_due_cost.py]: It's possible to obtain values to create from another function? So, other modules can extend dict informaton

File: l10n_it_ricevute_bancarie/models/account/account.py [hunk 4]
```diff
@@ -137,4 +142,99 @@ def _get_invoice_by_move_line(self, cr, uid, ids, context=None):
...
+        :param invoice_date_due: first date due of invoice
+        :param all_date_due: list of date of dues for partner
+        :return: True if month of invoice_date_due is in a list of all_date_due
+        """
+        check = False
+        for d in all_date_due:
+            if invoice_date_due[:7] == d[:7]:
+                check = True
```
Comment (OpenCode) [inline l10n_it_ricevute_bancarie/models/account/account.py]: Can you return immediately True to first true condition and return False as last possibility?

File: l10n_it_ricevute_bancarie/models/account/account.py [hunk 5]
```diff
@@ -137,4 +142,99 @@ def _get_invoice_by_move_line(self, cr, uid, ids, context=None):
...
+        return check
+
+    @api.multi
+    def action_move_create(self):
+        for invoice in self:
+            # ---- Add a line with payment cost for each due only for fist due
+            # ---- of the month
+            if invoice.type != 'out_invoice':
```
Comment (OpenCode) [inline l10n_it_ricevute_bancarie/models/account/account.py]: It it possible to have one if statement?

**Comment by andrea4ever** [comment]
Thank's @OpenCode for your review!

**Pull request merged**